base/collection_n1ql.go

```diff
@@ -119,8 +122,44 @@ func (c *Collection) CreatePrimaryIndex(indexName string, options *N1qlIndexOpti
 	return CreatePrimaryIndex(c, indexName, options)
 }
 
-func (c *Collection) WaitForIndexOnline(indexName string) error {
-	return WaitForIndexOnline(c, indexName)
+// WaitForIndexOnline takes set of indexes and watches them till they're online.
+func (c *Collection) WaitForIndexOnline(indexNames []string, watchPrimary bool) error {
+	waitChan := false
+	var waitErr error
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	cluster := c.cluster
+	mgr := cluster.QueryIndexes()
+
+	watchOption := gocb.WatchQueryIndexOptions{
+		WatchPrimary:   watchPrimary,
+		RetryStrategy:  gocb.NewBestEffortRetryStrategy(nil),
+		ScopeName:      c.ScopeName(),
+		CollectionName: c.Name(),
+	}
+
+	go func() {
```

Comment:
Rather than putting this function in a goroutnine, I think we can replace `db.waitForIndex` - we'll still call this `cluster.QueryIndexes.WatchIndexes` under the hood, but we want to call it for a really long time. 

I think we could do something like:

```
resultSet, resultsError := bucket.Query(queryStatement, nil, base.RequestPlus, true)

		// Immediately close results. We don't need these
		if resultSet != nil {
			resultSetCloseError := resultSet.Close()
			if resultSetCloseError != nil {
				base.InfofCtx(logCtx, base.KeyAll, "Failed to close query results when verifying index %q availability for bucket %q", base.MD(indexName), base.MD(bucket.GetName()))
			}
		}

		if resultsError == nil {
			return nil
		}
```

with `WatchIndexes` where `RetryStrategy` is `goCBv2FailFastRetryStrategy`